### PR TITLE
Optimize macro floorplanning

### PR DESCRIPTION
**Pull request by mkurc-ant**
The optimization include getter inlining and removal of `refer_lx_` and `refer_ly_` which were effectively unused but took part in checks in `SoftMacro::setX` and `SoftMacro::setY`.

The effect of the optimization was tested with `OpenROAD-flow-scripts` on `ariane133` design with `nandgate45` PDK. The stage `2_4_floorplan_macro` before the optimization took roughly **104** seconds whereas after the optimization about **83** seconds (on an Intel i7-8700 @ 3.20GHz CPU).

### CODE REVIEW

File: src/mpl2/src/object.h [hunk 1]
```diff
@@ -361,17 +361,17 @@ class HardMacro
   void setX(float x);
   void setY(float y);
   const std::pair<float, float> getLocation() const;
-  float getX() const;
-  float getY() const;
+  inline float getX() const {return x_;}
```
Comment (maliberty) [inline src/mpl2/src/object.h:364]: The inline keyword is unnecessary in this context, https://en.cppreference.com/w/cpp/language/inline:
```
A function defined entirely inside a class/struct/union definition, whether it's a member function or a non-member friend function, is implicitly an inline function
```

Please remove throughout this header

**Review by maliberty — CHANGES_REQUESTED** [review]
Thanks for the contribution

File: src/mpl2/src/object.h (hunk 1 above)
Comment (mkurc-ant) [reply src/mpl2/src/object.h]: Done.

File: src/mpl2/src/object.h [hunk 2]
```diff
@@ -547,13 +540,19 @@ class SoftMacro
...
-  float getWidth() const;
-  float getHeight() const;
+  float getX() const { return x_; }
+  float getY() const { return y_; }
+
+  // The position of pins relative to the lower left of the instance
+  float getPinX() const { return x_ + 0.5f * width_; }
+  float getPinY() const { return y_ + 0.5f * height_; }
+
+  const std::pair<float, float> getLocation() const
```
[the remark this answers is not in the document]
Comment (maliberty) [reply src/mpl2/src/object.h:550]: Please fix

[the remark this answers is not in the document]
Comment (mkurc-ant) [reply src/mpl2/src/object.h]: done

**Pull request merged**